app/models/registration_repository.rb

```diff
@@ -61,7 +61,41 @@ def self.store_search_query(session, query)
     end
   end
 
-  # Gets stored search query params
+  # restore search query to render search screen again
+  def self.restore_search_query_options(session)
+    session = session.dup
```

Comment:
there is no any particular reason, probably it's even required to clear the Session to avoid some side effects (e.g. user clears a value, but its old "converted" version from the previous search is alive)